File: jfingerprintmanager/src/main/java/com/jesusm/jfingerprintmanager/encryption/ui/FingerprintEncryptionDialogFragment.java
package com.jesusm.jfingerprintmanager.encryption.ui;

import android.support.annotation.NonNull;
import android.support.v4.hardware.fingerprint.FingerprintManagerCompat;

import com.jesusm.jfingerprintmanager.JFingerprintManager;
import com.jesusm.jfingerprintmanager.base.ui.FingerprintBaseDialogFragment;
import com.jesusm.jfingerprintmanager.encryption.presenter.FingerprintEncryptionDialogPresenter;

public class FingerprintEncryptionDialogFragment extends FingerprintBaseDialogFragment<FingerprintEncryptionDialogPresenter> implements FingerprintEncryptionDialogPresenter.View {

    private JFingerprintManager.EncryptionAuthenticatedCallback callback;

    @Override
    public void setCallback(@NonNull JFingerprintManager.FingerprintBaseCallback callback) {
        super.setCallback(callback);
        this.callback = (JFingerprintManager.EncryptionAuthenticatedCallback) callback;
    }

    @Override
    public void onAuthenticationSucceed(FingerprintManagerCompat.CryptoObject cryptoObject) {
        callback.onAuthenticationSuccess(cryptoObject);
    }

    public static class Builder extends FingerprintBaseDialogFragment.Builder<FingerprintEncryptionDialogFragment, FingerprintEncryptionDialogPresenter> {

        @Override
        protected FingerprintEncryptionDialogFragment createDialogFragment() {
            return new FingerprintEncryptionDialogFragment();
        }

        @Override
        protected FingerprintEncryptionDialogPresenter createPresenter(FingerprintEncryptionDialogFragment dialogFragment) {
            return new FingerprintEncryptionDialogPresenter(dialogFragment);
        }
    }
}

File: jfingerprintmanager/src/main/kotlin/com/jesusm/jfingerprintmanager/encryption/ui/FingerprintEncryptionDialogFragment.kt
package com.jesusm.jfingerprintmanager.encryption.ui

import android.support.v4.hardware.fingerprint.FingerprintManagerCompat
import com.jesusm.jfingerprintmanager.JFingerprintManager
import com.jesusm.jfingerprintmanager.base.ui.FingerprintBaseDialogFragment
import com.jesusm.jfingerprintmanager.encryption.presenter.FingerprintEncryptionDialogPresenter

class FingerprintEncryptionDialogFragment : FingerprintBaseDialogFragment<FingerprintEncryptionDialogPresenter>(), FingerprintEncryptionDialogPresenter.View {

    override fun onAuthenticationSucceed(cryptoObject: FingerprintManagerCompat.CryptoObject) {
        (callback as JFingerprintManager.EncryptionAuthenticatedCallback).onAuthenticationSuccess(cryptoObject)
    }

    class Builder : FingerprintBaseDialogFragment.Builder<FingerprintEncryptionDialogFragment, FingerprintEncryptionDialogPresenter>() {
        override fun addProperties(dialogFragment: FingerprintEncryptionDialogFragment) {

        }

        override fun createDialogFragment(): FingerprintEncryptionDialogFragment {
            return FingerprintEncryptionDialogFragment()
        }

        override fun createPresenter(view: FingerprintEncryptionDialogFragment): FingerprintEncryptionDialogPresenter {
            return FingerprintEncryptionDialogPresenter(view)
        }
    }
}